odl/tomo/data/mrc2014.py

```diff
@@ -0,0 +1,846 @@
+        modes.
+    kind : {'volume', 'projections'}
+        Interpretation of the 3D data, either as single 3D volume or as
+        a stack of 2D projections. The value is used for the ``'ispg'``
+        header entry.
+    extent : 3-sequence of floats, optional
+        Size of the 3D volume in meters. The values are used for
+        the ``'cella'`` header entry.
+        Default: ``num_cells``, resulting in ``(1, 1, 1)`` unit cells
+    axis_order : permutation of ``(0, 1, 2)`` optional
+        Order of the data axes as they should appear in the stored file.
+        The values are used for the ``'mapc', 'mapr', 'maps'`` header
+        entries.
+        Default: ``(0, 1, 2)``
+    dmin, dmax : float, optional
+        Minimum and maximum values of the data, used for header entries
+        ``'dmin'`` and ``'dmax'``, resp.
+        Default: 1.0, 0.0. These values indicate according to [Che+2015]_
+        that the values are considered as undetermined.
+    dmean, rms : float, optional
+        Mean and variance of the data, used for header entries ``'dmean'``
+        and ``'rms'``, resp.
+        Default: ``min(dmin, dmax) - 1, -1.0``. These values indicate
+        according to [Che+2015]_ that the values are considered as
+        undetermined.
+    mrc_version : 2-tuple of int, optional
+        Version identifier for the MRC file, used for the ``'nversion'``
+        header entry.
+        Default: ``(2014, 0)``
+    text_labels : sequence of strings
+        Maximal 10 strings with 80 characters each, used for the
+        ``'nlabl'`` and ``'label'`` header entries.
+        Default: ``[]``
+
+    Returns
+    -------
+    header : `OrderedDict`
+        Header stored in an ordered dictionary, where each entry has the
+        following form::
+
+            'name': {'value': value_as_array,
+                     'offset': offset_in_bytes
+                     'description': description_string}
+
+        All ``'value'``'s are `numpy.ndarray`'s with at least one
+        dimension.
+
+    References
+    ----------
+    [Che+2015] Cheng, A et al. *MRC2014: Extensions to the MRC format header
+    for electron cryo-microscopy and tomography*. Journal of Structural
+    Biology, 129 (2015), pp 146--150.
+    """
+    # Positional args
+    num_cells = [int(n) for n in num_cells]
+    kind, kind_in = str(kind).lower(), kind
+    if kind not in ('volume', 'projections'):
+        raise ValueError("`kind '{}' not understood".format(kind_in))
+
+    # Keyword args
+    extent = kwargs.pop('extent', num_cells)
+    axis_order = kwargs.pop('axis_order', (0, 1, 2))
+    if tuple(axis_order) not in permutations((0, 1, 2)):
+        raise ValueError('`axis_order` must be a permutation of (0, 1, 2), '
+                         'got {}'.format(axis_order))
+    dmin = kwargs.pop('dmin', 1.0)
+    dmax = kwargs.pop('dmax', 0.0)
+    dmean = kwargs.pop('dmean', min(dmin, dmax) - 1.0)
```

Comment:
It's according to the standard: If the mean value is smaller than both min and max, this signals "not defined".